here-naksha-app-service/src/test/java/com/here/naksha/app/service/UpdateFeatureTestHelper.java

```diff
@@ -0,0 +1,306 @@
+
+    // Then: Perform assertions
+    assertEquals(400, response.statusCode(), "ResCode mismatch");
+    JSONAssert.assertEquals(
+        "Update Feature error response doesn't match",
+        expectedBodyPart,
+        response.body(),
+        JSONCompareMode.LENIENT);
+    assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+  }
+
+  void tc0504_testUpdateFeaturesNoId() throws Exception {
+    // Test API : PUT /hub/spaces/{spaceId}/features
+    final String streamId = UUID.randomUUID().toString();
+
+    // Read request body
+    final String bodyJson = loadFileOrFail("TC0504_updateFeaturesNoIds/request.json");
+    // TODO: include geometry after Cursor-related changes ->
+    final Space space = parseJsonFileOrFail("TC0500_updateFeatures/create_space.json", Space.class);
+    // When: Create Features request is submitted to NakshaHub Space Storage instance
+    final HttpResponse<String> response =
+        nakshaClient.put("hub/spaces/" + space.getId() + "/features", bodyJson, streamId);
+
+    // Then: Perform assertions
+    assertEquals(200, response.statusCode(), "ResCode mismatch");
+    assertEquals(streamId, getHeader(response, HDR_STREAM_ID), "StreamId mismatch");
+
+    final XyzFeatureCollection responseFeatureCollection = parseJson(response.body(), XyzFeatureCollection.class);
+    Assertions.assertNotNull(responseFeatureCollection);
+    final List<String> inserted = responseFeatureCollection.getInserted();
+    Assertions.assertEquals(2, inserted.size());
+  }
+
+  void tc0505_testUpdateFeaturesWithUuid() throws Exception {
+    // Test API : PUT /hub/spaces/{spaceId}/features
+    final String streamId = UUID.randomUUID().toString();
+
+    // TODO: include geometry after Cursor-related changes ->
+    final Space space = parseJsonFileOrFail("TC0500_updateFeatures/create_space.json", Space.class);
+
+    final HttpResponse<String> getResponse =
+        nakshaClient.get("hub/spaces/" + space.getId() + "/features/my-custom-id-301-2", streamId);
+    final XyzFeature feature = parseJson(getResponse.body(), XyzFeature.class);
+    Assertions.assertNotNull(feature);
+    final XyzProperties newPropsOldUuid = feature.getProperties();
+    final XyzProperties newPropsOutdatedUuid = newPropsOldUuid.deepClone();
+    final XyzProperties nullUuidProps = new XyzProperties();
+    // Old UUID
+    newPropsOldUuid.put("speedLimit", "30");
+    // New UUID
+    newPropsOutdatedUuid.put("speedLimit", "120");
+    // Null UUID
+    nullUuidProps.put("uuid", null);
+    nullUuidProps.put("overriden", "yesyesyes");
+
+    // Execute request, correct UUID, should success
+    feature.setProperties(newPropsOldUuid);
+    final HttpResponse<String> responseUpdateSuccess = nakshaClient.put(
+        "hub/spaces/" + space.getId() + "/features",
+        """
+{
+"type": "FeatureCollection",
+"features": [
+""" + feature + "]}",
+        streamId);
```

Comment:
nth: consider indenting the multiline string, this looks a bit odd :D you can utilize `stripIndent` function - take a look at answers to this SO question: https://stackoverflow.com/questions/58030419/how-the-intents-processed-in-a-text-blockjava-13

Reply:
you're right it does look weird :) I ran the autoformat again for this file